docs/pages/kubernetes-access/discovery.mdx

```diff
@@ -0,0 +1,118 @@
+---
+title: Kubernetes Auto-Discovery (Preview)
+description: Detailed guides for configuring Kubernetes Auto-Discovery.
+layout: tocless-doc
+---
+
+Kubernetes Auto-Discovery is a Teleport feature that allows Kubernetes clusters 
+hosted on cloud providers to be discovered and enrolled automatically. 
+
+While discovering a new Kubernetes cluster, Teleport does not install any component 
+on the cluster. Instead, it requires direct access to the cluster's API and minimal access permissions.
+
+## How Kubernetes Auto-Discovery works
+
+Kubernetes Auto-Discovery consists of two distinct steps:
+
+### Polling cloud APIs
+
+The Teleport Discovery Service is responsible for scanning the configured cloud 
+providers and identifying if any Kubernetes clusters match a set of filtering labels. 
+When the process identifies a new Kubernetes cluster, it creates a dynamic 
+resource within Teleport. This resource includes information imported from the 
+cloud provider such as:
+
+- *Name*: Cluster name 
+- *Labels*
+  - Cluster tags.
+  - Cluster location.
+  - Identification of which cloud account the cluster belongs to — AWS Account ID / Azure Subscription ID.
+
+<Notice type="tip">
+You can import the cluster under a different name into Teleport's registry. 
+To achieve this, you must attach the following tag to the resources — EKS and AKS — in your cloud provider:
+ - ***key***: `teleport.dev/kubernetes-name` 
+ - ***value***: desired name 
+
+The Discovery Service will check if the cluster includes the tag and use its value 
+as the resource name in Teleport.
+
+You should use this feature whenever there are clusters in different regions/cloud providers 
+with the same name to prevent them from colliding in Teleport.
+
+</Notice>
+
+In addition to detecting new Kubernetes clusters, the Discovery Service also removes 
+— from Teleport's registry — the Kubernetes clusters that have been deleted or whose tags 
+no longer meet the filtering labels.
+
+The following snippet describes the different configuration options for the Discovery Service 
+and their default values.
+
+```yaml
+# This section configures the Discovery Service
+discovery_service:
+    enabled: "yes"
+    aws:
+       # AWS resource types. Valid options are:
+       # eks - discovers and registers AWS EKS clusters
+       # ec2 - discovers and registers AWS EC2 Machines
+     - types: ["eks"]
+       # AWS regions to search for resources from
+       regions: ["eu-central-1"]
```

Comment:
The comment above says these are default values - do we default to `eu-central-1` region? Based on our previous discussions with @xinding33 I think we should rather default to all US-based regions. Or just to empty and have user specify regions explicitly.

Reply:
EKS discovery uses the same existing AWS matcher that EC2 and databases use. It forces you to set values for regions and doesn't work with wildcards.
There are no default values, and you must specify the target region.

It will probably be good to extend wildcard support in the future.

This snippet was taken from EC2 discovery so it still includes `eu-central-1` (just like the original doc)